apps/api/src/app/shared/services/storage/storage.service.ts

```diff
@@ -80,6 +170,35 @@ export class AzureBlobStorageService implements StorageService {
     this.sharedKeyCredential
   );
 
+  async uploadFile(key: string, file: Buffer, contentType: string): Promise<PutObjectCommandOutput> {
+    const containerClient = this.blobServiceClient.getContainerClient(process.env.AZURE_CONTAINER_NAME);
+    const blockBlobClient = containerClient.getBlockBlobClient(key);
+
+    return (await blockBlobClient.upload(file, file.length, {
+      blobHTTPHeaders: {
+        blobContentType: contentType,
+      },
+    })) as unknown as PutObjectCommandOutput;
+  }
+
+  async getFile(key: string): Promise<Buffer> {
+    const containerClient = this.blobServiceClient.getContainerClient(process.env.AZURE_CONTAINER_NAME);
+    const blockBlobClient = containerClient.getBlockBlobClient(key);
+
+    const downloadBlockBlobResponse = await blockBlobClient.download(0);
+    if (downloadBlockBlobResponse.readableStreamBody === null) {
+      throw new NonExistingFileError();
+    }
+
+    return (await blockBlobClient.downloadToBuffer())[0] as unknown as Buffer;
+  }
```

Comment:
Please apply here the same error controlling pattern as in the AWS and GCP providers. 🙏🏻

Reply:
I misread the code.